### PR TITLE
Set up initial demo projects

### PR DESCRIPTION
**Pull request by octogonz**
This PR sets up some barebones TypeScript projects that we can use to demo the Rush configuration.  They aren't meant to provide a realistic JavaScript toolchain, but instead to demonstrate Rush features such as executing a binary from a non-published package, and building projects in the right order.

### CODE REVIEW

File: rush.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    {
+      "packageName": "my-controls",
+      "projectFolder": "libraries/my-controls",
+      "reviewCategory": "main"
+    },
+    {
+      "packageName": "my-toolchain",
+      "projectFolder": "tools/my-toolchain",
+      "reviewCategory": "tools"
+    }
```
Comment (nickpape-msft) [inline rush.json]: add publishing stuff in a comment

File: README.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+# Example Rush Monorepo
+
+This template is part of the documentation of the [Rush](https://rushjs.io/) tool.
+It contains templates for all the Rush configuration files, and a simple setup that
+builds two empty projects.
```
Comment (nickpape-msft) [inline README.md]: >two [](start = 7, length = 3)

three

File: libraries/my-controls/package.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  "license": "MIT",
+  "main": "lib/index.js",
+  "typings": "lib/index.d.ts",
+  "scripts": {
+    "build": "node_modules/.bin/my-build"
+  },
+  "dependencies": {},
+  "devDependencies": {
+    "my-toolchain": "^1.0.0",
+    "typescript": "^3.0.3"
```
Comment (nickpape-msft) [inline libraries/my-controls/package.json]: >typescript [](start = 5, length = 10)

is this necessary?

File: libraries/my-controls/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
+{
+  "$schema": "http://json.schemastore.org/tsconfig",
+
+  "compilerOptions": {
```
Comment (nickpape-msft) [inline libraries/my-controls/tsconfig.json]: >compilerOptions [](start = 3, length = 15)

could you move into the toolchain and use $extends?

File: common/config/rush/nonbrowser-approved-packages.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  "$schema": "https://developer.microsoft.com/json-schemas/rush/v5/approved-packages.schema.json",
+  "packages": [
+    {
+      "name": "colors",
+      "allowedCategories": [ "tools" ]
+    },
+    {
+      "name": "my-controls",
+      "allowedCategories": [ "main" ]
+    },
```
Comment (nickpape-msft) [inline common/config/rush/nonbrowser-approved-packages.json]: browser-approved

File: rush.json (hunk 1 above)
Comment (octogonz) [reply rush.json:39]: I'll do that in the next PR.

---
In reply to: [216845660](https://github.com/Microsoft/rush-example/pull/1#discussion_r216845660) [](ancestors = 216845660)

File: libraries/my-controls/package.json (hunk 3 above)
Comment (octogonz) [reply libraries/my-controls/package.json:14]: Not technically, but I think it's useful to illustrate a dependency used by all three projects

---
In reply to: [216846025](https://github.com/Microsoft/rush-example/pull/1#discussion_r216846025) [](ancestors = 216846025)

File: libraries/my-controls/tsconfig.json (hunk 4 above)
Comment (octogonz) [reply libraries/my-controls/tsconfig.json]: Good idea

---
In reply to: [216846178](https://github.com/Microsoft/rush-example/pull/1#discussion_r216846178) [](ancestors = 216846178)

**Review by iclanton — APPROVED** [review]
:shipit:

**Pull request merged**